### PR TITLE
Add alpha version of Keycloak

### PR DESCRIPTION
**Pull request by Js2604**
Add a FedStart compatible version of Keycloak. This chart has been tested successfully as an access management solution for other applications. However, the core Keycloak chart has yet to be configured to be highly available.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,36 +1,52 @@
 # Fedstart Helm Charts
 
-This repository provides reference implementations of open source helm charts with configurations that will work in Palantir's Fedstart environments. These helm chart configurations are intended to work in environments where cluster wide permissions are heavily restricted and/or unavailable.
+This repository provides reference implementations of open source helm charts with configurations that will work in
```
Comment (tabboud) [inline README.md]: Was there any particular reason the longer lines were trimmed in this file?

File: charts/alpha/keycloak/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Keycloak
+
+A Palantir Fedstart compliant helm-chart that
+configures [Keycloak](https://github.com/bitnami/charts/tree/main/bitnami/keycloak).
+
+## Configuration
+
+> Refer to the Grafana [README](https://github.com/bitnami/charts/tree/main/bitnami/keycloak#parameters) for all
```
Comment (tabboud) [inline charts/alpha/keycloak/README.md]: `s/Grafana/Keycloak`

File: charts/alpha/keycloak/values.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    existingSecret: "secret"
+    existingSecretDatabaseKey: ""
+    existingSecretHostKey: ""
+    existingSecretPasswordKey: ""
+    existingSecretPortKey: ""
+    existingSecretUserKey: ""
+
+  httpRelativePath: "/keycloak/"
+  containerSecurityContext:
+    enabled: false
```
Comment (tabboud) [inline charts/alpha/keycloak/values.yaml]: Why is this disabled? At the moment this fails helm-chart validation due to the security context.

File: README.md [hunk 4]
```diff
@@ -1,36 +1,52 @@
...
-* *Alpha* charts are works in progress, and have not passed internal Palantir testing. Users who are interested in deploying an Alpha chart should reach out to Palantir to provide details about your use case. Alpha charts are organized under the `charts/alpha/<name of helm chart>` directory.
+* *Generally Available (GA)* charts are deployed by one or more Palantir Fedstart partners and are expected to be
+  production ready. Users can use these at their own discretion, with no need to ask Palantir before doing so. Generally
+  Available charts are organized under the `charts/ga/<name of helm chart>` directory.
+* *Beta* charts have been tested internally and are ready to be tested by Fedstart partners. Users who are interested in
+  deploying a Beta chart should reach out to Palantir before doing so. Beta charts are organized under
+  the `charts/beta/<name of helm chart>` directory.
+* *Alpha* charts are works in progress, and have not passed internal Palantir testing. Users who are interested in
+  deploying an Alpha chart should reach out to Palantir to provide details about your use case. Alpha charts are
+  organized under the `charts/alpha/<name of helm chart>` directory.
```
Comment (ashrayjain) [inline README.md]: Please remove all these unrelated changes or at a minimum, open a separate PR for these if you still want these.

File: charts/alpha/keycloak/values.yaml (hunk 3 above)
Comment (Js2604) [reply charts/alpha/keycloak/values.yaml]: I had disabled this because the podsecuritycontext gets injected when deployed into rubix. I can explicitly add to avoid blocking the publish flow though.

File: charts/alpha/keycloak/values.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+keycloak:
+  networkPolicy:
+    enabled: false
+
+  postgresql:
+    enabled: false
+    primary:
```
Comment (ashrayjain) [inline charts/alpha/keycloak/values.yaml]: since we have enabled: false here already, do we actually need the `primary` overrides? My read is that the sub chart will be entirely disabled because of `enabled: false`

File: charts/alpha/keycloak/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+|-----------------------------------|------------------------------------------------------------------------------------|----------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `postgresql.enabled`              | Switch to true to enable the PostgreSQL helm chart instead of using an external DB | `false`  | To ensure your Keycloak deployment is highly available, it's recommended that you use an external DB instance. Set up this connection using the overrides under the [`externalDatabase`](https://github.com/bitnami/charts/tree/main/bitnami/keycloak#database-parameters) parameter. |
```
Comment (ashrayjain) [inline charts/alpha/keycloak/README.md]: do we need to document this? Do we actually want to support using postgresql at all?

File: charts/alpha/keycloak/values.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      annotations:
+        com.palantir.rubix.service/pod-cert: '{}'
+    annotations:
+      com.palantir.rubix.service/pod-cert: '{}'
+
+  tls:
+    enabled: true
+    existingSecret: "cert-{{ .Release.Name }}"
+    truststoreFilename: truststore.jks
+    keystoreFilename: keystore.jks
```
Comment (ashrayjain) [inline charts/alpha/keycloak/values.yaml:37]: we are moving towards opt-in generation of JKS so lets update our use of the cert-api to explicitly request JKS. Let's flup internally on this one.

**Review by ashrayjain — APPROVED** [review]
